lib/livebook_web/live/session_live/k8s_runtime_component.ex

```diff
@@ -0,0 +1,952 @@
+defmodule LivebookWeb.SessionLive.K8sRuntimeComponent do
+  use LivebookWeb, :live_component
+
+  import Ecto.Changeset
+
+  alias Livebook.{Session, Runtime}
+  alias Livebook.K8s.{Auth, Pod, PVC}
+
+  @config_secret_prefix "K8S_RUNTIME_"
+  @kubeconfig_pipeline Application.compile_env(:livebook, :k8s_kubeconfig_pipeline)
+
+  @impl true
+  def mount(socket) do
+    unless Livebook.Config.runtime_enabled?(Livebook.Runtime.K8s) do
+      raise "runtime module not allowed"
+    end
+
+    kubeconfig = Kubereq.Kubeconfig.load(@kubeconfig_pipeline)
+    context_options = Enum.map(kubeconfig.contexts, & &1["name"])
+
+    {:ok,
+     socket
+     |> assign(
+       kubeconfig: kubeconfig,
+       context_options: context_options,
+       context: nil,
+       reqs: nil,
+       cluster_check: %{status: :initial, error: nil},
+       namespace: nil,
+       namespace_options: nil,
+       rbac: %{status: :inflight, errors: [], permissions: []},
+       save_config: nil,
+       pvcs: nil,
+       pvc_action: nil,
+       home_pvc: nil,
+       docker_tag: hd(Livebook.Config.docker_images()).tag,
+       pod_template: %{template: Pod.default_pod_template(), status: :valid, message: nil}
+     )
+     |> set_context(kubeconfig.current_context)}
```

Comment:
In case a k8s runtime is already configured with a different context, we will try to load it immediately in `update`. So I would move this call to `update`, I think the default `cond` clause?